src/core_plugins/metrics/public/components/vis_editor_visualization.js

```diff
@@ -64,6 +66,24 @@ class VisEditorVisualization extends Component {
     window.removeEventListener('mouseup', this.handleMouseUp);
   }
 
+  _loadVisualization() {
+    getVisualizeLoader().then(loader => {
+      this._loader = loader;
+      this._loader.embedVisualizationWithSavedObject(this._visEl.current, this.props.savedObj, {
```

Comment:
I think that actually can cause an error, in case we're already unmounted at the time the `loader` is there. I am not exactly sure if we can check the ref for that state (@markov00 any idea?). Otherwise we'll need to set a flag in `componentWillUnmount` that will prevent this method from actually being called.

Reply:
You are right, I think we can directly check if `this._visEl.current` is `null` in the promise chain.